shells/pipes-shell/source/verbs/run-arc.js

```diff
@@ -14,10 +14,39 @@ import {devtoolsArcInspectorFactory} from '../../../../build/devtools-connector/
 import {Runtime} from '../../../../build/runtime/runtime.js';
 import {portIndustry} from '../pec-port.js';
 
-const {log, warn} = logsFactory('pipe');
+const {log, warn} = logsFactory('runArc');
 
-// The implementation was forked from verbs/spawn.js
-export const runArc = async (msg, bus, runtime, defaultStorageKeyPrefix) => {
+const runQueue = [];
+
+// TODO(sjmiles): serialize requests to runArc to reduce synchronization
+// burden on apps. Revisit at some point.
+
+export const runArc = async (...args) => {
+  if (runArc.busy) {
+    runQueue.push(args);
+    warn('queue.push: length is ', runQueue.length);
+  } else {
+    await _runArc(...args);
+    warn('queue.shift: length is ', runQueue.length);
+    const nextArgs = runQueue.shift();
+    if (nextArgs) {
+      await runArc(...nextArgs);
```

Comment:
Should you await here? if you do, then promise returned by runArc represents depleting an entire queue, which is probably not what we want. WDYT?

Reply:
Argh, good catch. I knew it was too easy :) ...